src/java.base/share/native/libjli/java.c

```diff
@@ -326,6 +329,33 @@ JLI_Launch(int argc, char ** argv,              /* main argc, argv */
         }
     }
 
+    {
+        /* Process -Xmso in the OPENJ9_JAVA_OPTIONS environment variable to
+         * set the main thread stack size. May be overridden by a later command
+         * line option.
+         */
+        JLI_List openj9Args = JLI_List_new(8); /* 8 is arbitrary */
+        if (JLI_ParseOpenJ9ArgsFromEnvVar(openj9Args, "OPENJ9_JAVA_OPTIONS")) {
+            jboolean foundMso = JNI_FALSE;
+            size_t i = openj9Args->size;
+            for (; i > 0; i--) {
+                if (!foundMso && JLI_StrCCmp(openj9Args->elements[i - 1], "-Xmso") == 0) {
+                    jlong tmp = 0;
+                    if (parse_size(openj9Args->elements[i - 1] + 5, &tmp)) {
+                        threadStackSize = tmp;
+                        if (threadStackSize > 0 && threadStackSize < (jlong)STACK_SIZE_MINIMUM) {
+                            threadStackSize = STACK_SIZE_MINIMUM;
+                        }
+                    }
+                    foundMso = JNI_TRUE;
+                }
+                JLI_MemFree(openj9Args->elements[i - 1]);
+            }
+            JLI_MemFree(openj9Args->elements);
+            JLI_MemFree(openj9Args);
```

Comment:
Suggestions:
* decrement `i` at the beginning of the loop rather than use `i - 1` several places
* use `JLI_List_free()` instead of essentially repeating its implementation here via `JLI_MemFree()`
* remove `foundMso` and just say `break;` after finding `-Xmso`

```c
            size_t i = openj9Args->size;
            while (i > 0) {
                i -= 1;
                if (JLI_StrCCmp(openj9Args->elements[i], "-Xmso") == 0) {
                    jlong tmp = 0;
                    fprintf(stderr, "Found thread stack size: '%s'\n", openj9Args->elements[i]);
                    if (parse_size(openj9Args->elements[i] + 5, &tmp)) {
                        threadStackSize = tmp;
                        if (threadStackSize > 0 && threadStackSize < (jlong)STACK_SIZE_MINIMUM) {
                            threadStackSize = STACK_SIZE_MINIMUM;
                        }
                    }
                    break;
                }
            }
            JLI_List_free(openj9Args);
```